taxi/views.py

```diff
@@ -61,12 +65,30 @@ class CarListView(LoginRequiredMixin, generic.ListView):
 class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
 
+    def post(
+            self, request: HttpRequest, *args, **kwargs
+    ) -> HttpResponseRedirect:
+        car = self.get_object()
+        driver = request.user
+
+        if "assign" in request.POST:
+            car.drivers.add(driver)
+        elif "remove" in request.POST:
+            car.drivers.remove(driver)
+
+        return redirect("taxi:car-detail", pk=car.pk)
+
 
 class CarCreateView(LoginRequiredMixin, generic.CreateView):
     model = Car
     fields = "__all__"
     success_url = reverse_lazy("taxi:car-list")
 
+    def get_form(self, *args, **kwargs) -> forms.ModelForm:
+        form = super().get_form(*args, **kwargs)
+        form.fields["drivers"].widget = forms.CheckboxSelectMultiple()
```

Comment:
It works but your code would be better structured if you create a Form where you override widgets